library/channel.c

```diff
@@ -210,8 +216,15 @@ int ziti_close_channels(struct ziti_ctx *ztx, int err) {
 
 static void on_tls_close(uv_handle_t *s) {
     tlsuv_stream_t *tls = (tlsuv_stream_t *) s;
+    ziti_channel_t *ch = tls->data;
+    ch->connection = NULL;
```

Comment:
is there another reference to the connection? this looks like another leak.

Reply:
handle(`tls`) is the same as `ch->connection`